### PR TITLE
Add arguments to exception struct

### PR DESCRIPTION
**Pull request by nitinstp23**
@vasuadari Can you review this?

### CODE REVIEW

**Comment by vasuadari** [comment]
@vasuadari Checking it.

File: lib/messengers/http_messenger.ex [hunk 1]
```diff
@@ -106,7 +107,8 @@ defmodule TxtLocalEx.HttpMessenger do
 
     case Request.request(:post, @message_status_path, sms_payload) do
       {:ok, response} -> response.body
-      {:error, error} -> raise TxtLocalEx.Errors.ApiError, error.reason
+      {:error, error} when is_binary(error) -> raise TxtLocalEx.Errors.ApiError, error
+      {:error, error} -> raise error
```
Comment (vasuadari) [inline lib/messengers/http_messenger.ex:109]: Can you write a test to cover this?

**Review by vasuadari — COMMENTED** [review]
Changes looks good other than a missing test.

File: lib/messengers/http_messenger.ex (hunk 1 above)
Comment (nitinstp23) [reply lib/messengers/http_messenger.ex:109]: We don't need to handle this clause - 

```elixir
{:error, error} when is_binary(error) -> raise TxtLocalEx.Errors.ApiError, error
```

Removing it!

**Pull request merged**